packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1212,9 +1213,28 @@ function detachFiberMutation(fiber: Fiber) {
   fiber.return = null;
 }
 
-export function detachFiberAfterEffects(fiber: Fiber): void {
+export function detachFiberAfterEffects(
+  fiber: Fiber,
+  recurseIntoSibbling: ?boolean,
+): void {
   // Null out fields to improve GC for references that may be lingering (e.g. DevTools).
   // Note that we already cleared the return pointer in detachFiberMutation().
+  if (enableStrongMemoryCleanup) {
+    if (fiber.child) {
+      detachFiberAfterEffects(fiber.child, true);
+    }
+    if (fiber.sibling && recurseIntoSibbling === true) {
+      detachFiberAfterEffects(fiber.sibling, true);
+    }
+    if (fiber.stateNode) {
+      Object.keys(fiber.stateNode).forEach(key => {
+        if (key.indexOf('__react') === 0) {
+          delete fiber.stateNode[key];
+        }
+      });
+    }
```

Comment:
This feels pretty heavy. Why is it needed given that we null out `fiber.stateNode` below (and we null out all expensive fiber fields)?

Even if something has a reference to the fiber at this point, it wouldn't have any deep references.

Reply:
I agree. We’re starting with the shotgun approach to confirm if this kills the leak. If not it doesn’t matter, we have to keep looking and narrow down what we miss.

When the leak is fixed then we can come back and find the right way to do this clean up.

I think you are correct. Worse case keeping a ref to an unmourned / detached DOM mode will be O(1) by retaining 1 and only 1 associated fiber node. Well sort this out once we confirm the leak is gone.